sdk/eventhub/event-hubs/test/node/disconnects.spec.ts

```diff
@@ -55,25 +55,65 @@ describe("disconnected", function() {
         service.connectionString,
         service.path
       );
+
+      const producer = new EventHubProducerClient(service.connectionString, service.path);
+      const eventsToSend: EventData[] = [{ body: "the first event" }, { body: "the second event" }];
+
+      const maxWaitTimeInSeconds = 10;
       const partitionId = "0";
       const partitionProperties = await client.getPartitionProperties(partitionId);
       const clientConnectionContext = client["_context"];
 
+      // Send the first event after getting partition properties so that we can expect to receive it.
+      await producer.sendBatch([eventsToSend[0]], { partitionId });
+
       let subscription: Subscription | undefined;
       let originalConnectionId: string;
 
+      let processEventsInvocationCount = 0;
+      let firstInvocationEndTime = 0;
       await new Promise((resolve, reject) => {
         subscription = client.subscribe(
           partitionId,
           {
             processEvents: async (data) => {
+              processEventsInvocationCount++;
               should.exist(data);
-              should.equal(data.length, 0);
-              if (!originalConnectionId) {
+              if (processEventsInvocationCount === 1) {
+                should.equal(
+                  data.length,
+                  1,
+                  "Expected to receive 1 event in first processEvents invocation."
+                );
+                should.equal(data[0].body, eventsToSend[0].body);
                 originalConnectionId = clientConnectionContext.connectionId;
                 // Trigger a disconnect on the underlying connection.
                 clientConnectionContext.connection["_connection"].idle();
-              } else {
+                firstInvocationEndTime = Date.now();
+              } else if (processEventsInvocationCount === 2) {
+                // No new events should have been received at this point.
+                should.equal(
+                  data.length,
+                  0,
+                  "Expected to receive 0 events in second processEvents invocation."
+                );
+                // The elapsed time since the last processEvents invocation should be >= maxWaitTimeInSeconds
+                should.equal(
+                  Date.now() - firstInvocationEndTime >= maxWaitTimeInSeconds,
+                  true,
+                  "Expected elapsed time between first and second processEvents invocations to be >= maxWaitTimeInSeconds."
+                );
+                const newConnectionId = clientConnectionContext.connectionId;
+                should.not.equal(originalConnectionId, newConnectionId);
+                // Send a new event that will be immediately receivable.
+                await producer.sendBatch([eventsToSend[1]], { partitionId });
```

Comment:
So technically if we had sent this after the triggered disconnect above (maybe you're just trying to wrestle with timing) then we wouldn't have an "empty" iteration here for processEventInvocation === 2?

I think a bit more commentary at the top of this set of checks would help explain the flow (and why).

Reply:
I wanted to prevent a flaky test scenario by sending the 2nd event only after the 2nd processEvents invocation.

If I had done it in the 1st processEvents invocation, then depending on latency/how fast the service processed the event it would be possible to receive it before the underlying receiver was closed. That would cause the 2nd processEvents to return as soon as it saw the event.

I'll try to make the comments clearer on what's being tested.